packages/user-operation-controller/src/UserOperationController.ts

```diff
@@ -537,6 +537,15 @@ export class UserOperationController extends BaseController<
     validateUpdateUserOperationResponse(response);
 
     userOperation.paymasterAndData = response.paymasterAndData ?? EMPTY_BYTES;
+    if (response.callGasLimit) {
```

Comment:
We'll need to update the `SnapSmartContractAccount` to ensure we return these values from the `KeyringController`, though I assume that would also require an updated `KeyringController`?

Reply:
Yes, we will also need to update the keyring controller